samples/drivers/jesd216/boards/nrf52840dk_nrf52840_spi.overlay

```diff
@@ -1,38 +1,26 @@
 /*
- * Copyright (c) 2022 Nordic Semiconductor ASA
+ * Copyright (c) 2022-2023 Nordic Semiconductor ASA
  *
  * SPDX-License-Identifier: Apache-2.0
  */
 
-/* This DTS overlay allows to test MX25R6435F  using SPI_NOR driver */
-
-&pinctrl {
-	spi1_alt_default: spi1_alt_default {
-		group1 {
-			psels = <NRF_PSEL(SPIM_SCK, 0, 19)>,
-				<NRF_PSEL(SPIM_MOSI, 0, 20)>,
-				<NRF_PSEL(SPIM_MISO, 0, 21)>;
-		};
-	};
+/delete-node/ &mx25r64;
 
-	spi1_alt_sleep: spi1_alt_sleep {
-		group1 {
-			psels = <NRF_PSEL(SPIM_SCK, 0, 19)>,
-				<NRF_PSEL(SPIM_MOSI, 0, 20)>,
-				<NRF_PSEL(SPIM_MISO, 0, 21)>;
-			low-power-enable;
-		};
-	};
+&qspi {
+	status = "disabled";
 };
 
-/delete-node/ &mx25r64;
-
-&spi1 {
+/* The mx25, on nrf52840dk_nrf52840, uses pins for spi0, spi1, spi2 and spi3
+ * to provide quad-spi feature. In individual specifications each of the spi
+ * notes define own clock source (SCK), but spi2 shares the same clock source
+ * as qspi configuration, which is pin (0,19). That is why spi2 is used here
+ * to communicate with mx25, when qspi is not used, to avoid rerouting clock
+ * pin.
+ */
+&spi2 {
 	compatible = "nordic,nrf-spi";
 	status = "okay";
 	cs-gpios = <&gpio0 17 GPIO_ACTIVE_LOW>;
-	pinctrl-0 = <&spi1_alt_default>;
-	pinctrl-1 = <&spi1_alt_sleep>;
 	pinctrl-names = "default", "sleep";
 	mx25r64: mx25r6435f@0  {
 		compatible = "jedec,spi-nor";
```

Comment:
Max SPIM3 speed is 32MHz but part below specifies 33MHz, for SPIM2 max speed is 8MHz. Also to note is the speed of the QSPI part itself, there are 2 modes - high performance (something like 96MHz max frequency) and ultra low power mode (8MHz max frequency), the made is changed with a volatile bit, the default depends upon which part number chip is used, I'm unsure which one is used on the DK but would want to check that

Reply:
The SPIM2 max is 8Mbit so 8MHz would be max, will fix.
I should also put the
```
        pinctrl-0 = <&spi2_default>;
        pinctrl-1 = <&spi2_sleep>;
```
in place of removed SPI lines here.